kv/kvclient/rangefeed: TestBackoffOnRangefeedFailure failed

kv/kvclient/rangefeed.TestBackoffOnRangefeedFailure [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8857971?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8857971?buildTab=artifacts#/) on master @ [e05c9c1ef273daca8b48b82075b0a79a386c3d95](https://github.com/cockroachdb/cockroach/commits/e05c9c1ef273daca8b48b82075b0a79a386c3d95):


```
=== RUN   TestBackoffOnRangefeedFailure
W230227 16:10:09.110409 19502 kv/kvclient/rangefeed/scanner.go:70  [T1,rangefeed=foo] 2318  failed to perform initial scan: scan failed
    rangefeed.go:312: Unexpected call to *rangefeed.MockDB.RangeFeed([context.Background.WithCancel.WithValue(type logtags.contextLogTagsKey, val rangefeed=foo).WithCancel.WithValue(type tracing.activeSpanKey, val rangefeed).WithValue(type pprof.labelContextKey, val {"rangefeed":"foo"}).WithCancel [/M{in-ax}] 0,0 false 0x400296cd20 0x17c82d0]) at github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed/rangefeed.go:312 because: 
        expected call at github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed_test/pkg/kv/kvclient/rangefeed/rangefeed_mock_test.go:365 has the wrong number of arguments. Got: 6, want: 5
        expected call at github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed_test/pkg/kv/kvclient/rangefeed/rangefeed_mock_test.go:368 has the wrong number of arguments. Got: 6, want: 5
    rangefeed_mock_test.go:388: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/9260/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/kvclient/rangefeed/rangefeed_test_/rangefeed_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvclient/rangefeed/rangefeed_mock_test.go:388
        	Error:      	timed out waiting for retries
        	Test:       	TestBackoffOnRangefeedFailure
    controller.go:137: missing call(s) to *rangefeed.MockDB.RangeFeed(is anything, is anything, is anything, is anything, is anything) github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed_test/pkg/kv/kvclient/rangefeed/rangefeed_mock_test.go:365
    controller.go:137: missing call(s) to *rangefeed.MockDB.RangeFeed(is anything, is anything, is anything, is anything, is anything) github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed_test/pkg/kv/kvclient/rangefeed/rangefeed_mock_test.go:368
    controller.go:137: aborting test due to missing call(s)
--- FAIL: TestBackoffOnRangefeedFailure (10.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/replication
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackoffOnRangefeedFailure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-24845

kv/kvclient/rangefeed.TestBackoffOnRangefeedFailure [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8860219?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8860219?buildTab=artifacts#/) on master @ [7730070ad3840004c3cf65a035fee7298e372548](https://github.com/cockroachdb/cockroach/commits/7730070ad3840004c3cf65a035fee7298e372548):


```
=== RUN   TestBackoffOnRangefeedFailure
W230227 18:38:50.066058 19387 kv/kvclient/rangefeed/scanner.go:70  [T1,rangefeed=foo] 2320  failed to perform initial scan: scan failed
    rangefeed.go:312: Unexpected call to *rangefeed.MockDB.RangeFeed([context.Background.WithCancel.WithValue(type logtags.contextLogTagsKey, val rangefeed=foo).WithCancel.WithValue(type tracing.activeSpanKey, val rangefeed).WithValue(type pprof.labelContextKey, val {"rangefeed":"foo"}).WithCancel [/M{in-ax}] 0,0 false 0x40019c8840 0x17c82d0]) at github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed/rangefeed.go:312 because: 
        expected call at github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed_test/pkg/kv/kvclient/rangefeed/rangefeed_mock_test.go:365 has the wrong number of arguments. Got: 6, want: 5
        expected call at github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed_test/pkg/kv/kvclient/rangefeed/rangefeed_mock_test.go:368 has the wrong number of arguments. Got: 6, want: 5
    rangefeed_mock_test.go:388: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/8637/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/kv/kvclient/rangefeed/rangefeed_test_/rangefeed_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvclient/rangefeed/rangefeed_mock_test.go:388
        	Error:      	timed out waiting for retries
        	Test:       	TestBackoffOnRangefeedFailure
    controller.go:137: missing call(s) to *rangefeed.MockDB.RangeFeed(is anything, is anything, is anything, is anything, is anything) github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed_test/pkg/kv/kvclient/rangefeed/rangefeed_mock_test.go:365
    controller.go:137: missing call(s) to *rangefeed.MockDB.RangeFeed(is anything, is anything, is anything, is anything, is anything) github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed_test/pkg/kv/kvclient/rangefeed/rangefeed_mock_test.go:368
    controller.go:137: aborting test due to missing call(s)
--- FAIL: TestBackoffOnRangefeedFailure (10.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackoffOnRangefeedFailure.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvclient/rangefeed/BUILD.bazel
+++ b/pkg/kv/kvclient/rangefeed/BUILD.bazel
@@ -83,7 +83,6 @@ go_test(
         "//pkg/storage",
         "//pkg/testutils",
         "//pkg/testutils/serverutils",
-        "//pkg/testutils/skip",
         "//pkg/testutils/sqlutils",
         "//pkg/testutils/storageutils",
         "//pkg/testutils/testcluster",

--- a/pkg/kv/kvclient/rangefeed/rangefeed_mock_test.go
+++ b/pkg/kv/kvclient/rangefeed/rangefeed_mock_test.go
@@ -20,7 +20,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/kv/kvclient/rangefeed"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvpb"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/retry"
@@ -350,8 +349,6 @@ func TestRangeFeedMock(t *testing.T) {
 func TestBackoffOnRangefeedFailure(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 
-	skip.WithIssue(t, 97725, "flaky test")
-
 	ctx, cancel := context.WithCancel(context.Background())
 	stopper := stop.NewStopper()
 	defer stopper.Stop(ctx)
@@ -365,10 +362,10 @@ func TestBackoffOnRangefeedFailure(t *testing.T) {
 
 	// Make sure rangefeed is retried even after 3 failures, then succeed and cancel context
 	// (which signals the rangefeed to shut down gracefully).
-	db.EXPECT().RangeFeed(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).
+	db.EXPECT().RangeFeed(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).
 		Times(3).
 		Return(errors.New("rangefeed failed"))
-	db.EXPECT().RangeFeed(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).
+	db.EXPECT().RangeFeed(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).
 		Do(func(context.Context, []roachpb.Span, hlc.Timestamp, bool, chan<- kvcoord.RangeFeedMessage, ...kvcoord.RangeFeedOption) {
 			cancel()
 		}).

